scripts/airgapped/get-all-images.py

```diff
@@ -0,0 +1,177 @@
+import argparse
+import logging
+import shutil
+import subprocess
+
+import git
+import yaml
+
+# logging
+LOG_FORMAT = "%(levelname)s \t| %(message)s"
+logging.basicConfig(format=LOG_FORMAT, level=logging.INFO)
+log = logging.getLogger(__name__)
+
+# consts
+EXCLUDE_REPOS = ["kubeflow-roles"]
+
+GH_REPO_KEY = "_github_repo_name"
+GH_BRANCH_KEY = "_github_repo_branch"
+GH_DEPENDENCY_REPO_KEY = "_github_dependency_repo_name"
+GH_DEPENDENCY_BRANCH_KEY = "_github_dependency_repo_branch"
+GET_IMAGES_SH = "tools/get-images.sh"
+
+AIRGAP_TESTING_IMAGES = [
+    "charmedkubeflow/pipelines-runner:ckf-1.8",
+    "docker.io/kubeflowkatib/simple-pbt:v0.16.0",
+    "ghcr.io/knative/helloworld-go:latest",
+    "gcr.io/kubeflow-ci/tf-mnist-with-summaries:1.0",
+]
+
+
+def is_dependency_app(app: dict) -> bool:
+    """Detect if app in bundle is not owned by Analytics team."""
+    if GH_DEPENDENCY_REPO_KEY in app and GH_DEPENDENCY_BRANCH_KEY in app:
+        return True
+
+    return False
+
+
+def bundle_app_contains_gh_metadata(app: dict) -> bool:
+    """
+    Given an application in a bundle check if it contains github metadata keys
+    to be able to be parsed properly.
+    """
+    if is_dependency_app(app):
+        return True
+
+    if GH_REPO_KEY in app and GH_BRANCH_KEY in app:
+        return True
+
+    return False
+
+
+def clone_repo(repo_name: str, branch: str) -> str:
+    """Clones locally a repo and returns the folder created."""
+    repo_url = f"https://github.com/canonical/{repo_name}.git"
+
+    logging.info(f"Cloning repo {repo_url}")
+    repo = git.Repo.clone_from(repo_url, str(repo_name))
+
+    logging.info(f"Checking out to branch {branch}")
+    repo.git.checkout(branch)
+
+    return repo_name
```

Comment:
I updated the code to instead return the path of the cloned repo. Although the code doesn't explicitly do something to change it from the argument of `repo_name` I still like this return value since:
1. It makes it clear what the function returns
2. It returns the full path 

34f7602

Reply:
> It makes it clear what the function returns

It can return None and that is totally fine as well, right?

> It returns the full path

I think that we know the path before hand and can process it even w/o this function that clones a repo. You can have the full path already. I guess the way to look at it is: this method is cloning a repository in your local device, rather than it processes the path you give to it as an arg.
If you really think returning the full path is necessary, I'd suggest you check for the cloned repo and verify that the path actually exists and has the repo you intended to clone. Maybe with these checks it could be more useful.

Reply:
c43b06a